File: service/src/main/java/org/cadadr/conjugaison/DataInitializer.java
package org.cadadr.conjugaison;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.cadadr.conjugaison.domain.VerbConjugation;
import org.cadadr.conjugaison.service.VerbConjugationService;
import org.jetbrains.annotations.NotNull;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.io.File;
import java.io.IOException;
import java.util.List;

@Component
public class DataInitializer {

    private final Logger logger = LoggerFactory.getLogger(DataInitializer.class);

    private static final File DATA_FILE = new File("verbs.json");

    private final VerbConjugationService verbConjugationService;

    @Autowired
    public DataInitializer(@NotNull VerbConjugationService verbConjugationService) {
        this.verbConjugationService = verbConjugationService;
        init();
    }

    private void init() {
        logger.info("Saving data to MongoDB");
        verbConjugationService.saveAll(readData());
    }

    private @NotNull List<VerbConjugation> readData() {
        final ObjectMapper mapper = new ObjectMapper();

        try {
            final List<VerbConjugation> result = mapper.readValue(DATA_FILE, new TypeReference<>() {});
            logger.info("Read {} entries from '{}'", result.size(), DATA_FILE.getName());
            return result;
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

}

File: service/src/main/kotlin/org/cadadr/conjugaison/DataInitializer.kt
package org.cadadr.conjugaison

import kotlinx.serialization.decodeFromString
import kotlinx.serialization.json.Json
import org.cadadr.conjugaison.domain.VerbConjugation
import org.cadadr.conjugaison.service.VerbConjugationService
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component
import java.io.File

@Component
class DataInitializer(
    private val verbConjugationService: VerbConjugationService
) {
    private val logger = LoggerFactory.getLogger(DataInitializer::class.java)
    private val DATA_FILE = File("verbs.json")

    init {
        init()
    }

    private fun init() {
        logger.info("Saving data to MongoDB")
        verbConjugationService.saveAll(readData())
    }

    private fun readData(): List<VerbConjugation> {
        val conjugations: List<VerbConjugation> = Json.decodeFromString(DATA_FILE.readText())
        logger.info("Read {} entries from '{}'", conjugations.size, DATA_FILE.name)
        return conjugations
    }

}
